client/rpc/src/eth/execute.rs

```diff
@@ -112,7 +112,11 @@ where
 			}
 		};
 
-		if let Ok(BlockStatus::Unknown) = self.client.status(id) {
+		let Ok(hash) = self.client.expect_block_hash_from_id(&id) else {
+			return Err(crate::err(JSON_RPC_ERROR_DEFAULT, "block not found", None));
```

Comment:
This causes a [failure](https://github.com/paritytech/frontier/pull/980/checks#step:9:254) in integration tests. Perhaps the test case should be updated and/or a new one added to check this part of the code.

Reply:
I think it's ok to return "header not found" here, and remove `.client.status` call later, as expect_block_hash_from_id already checks block existence.